source/common/milestone_tracker_impl.h

```diff
@@ -16,13 +19,30 @@ class MilestoneTrackerImpl : public MilestoneTracker {
   uint32_t registerMilestone(absl::string_view name) override;
   void markMilestone(const uint32_t milestone) override;
   const Envoy::MonotonicTime getMilestone(const uint32_t milestone) const override;
-  std::chrono::duration<double> elapsedBetween(const uint32_t from,
-                                               const uint32_t to) const override;
+  std::chrono::nanoseconds elapsedBetween(const uint32_t from, const uint32_t to) const override;
 
 private:
   Envoy::TimeSource& time_source_;
   std::vector<std::tuple<Envoy::MonotonicTime, std::string>> timestamps_;
   int last_milestone_{-1}; // for verification only.
 };
 
+// Compose a poolable MilestoneTrackerImpl.
+class PoolableMilestoneTrackerImpl : public MilestoneTrackerImpl, public PoolableImpl {
+public:
+  PoolableMilestoneTrackerImpl(Envoy::TimeSource& time_source)
+      : MilestoneTrackerImpl(time_source) {}
+};
+
+// Declare a pool for the poolable milestone
+class MilestoneTrackerPoolImpl : public PoolImpl<PoolableMilestoneTrackerImpl> {
```

Comment:
Same comment as for `StreamDecoderPoolImpl`, I don't think this needs to be a class.

Reply:
fixed!